spec/util/throwUnobservableError-spec.ts

```diff
@@ -0,0 +1,67 @@
+import { expect } from 'chai';
+import { createInvalidObservableTypeError } from '../../src/internal/util/throwUnobservableError';
+import { from } from 'rxjs';
+
+/** @test {createInvalidObservableTypeError} */
+describe('createInvalidObservableTypeError', () => {
+  /** JSON types **/
+  it('Should handle a string', () => {
+    const error = createInvalidObservableTypeError('FooBar');
+    expect(error.message).to.match(/^You provided 'FooBar' where a stream was expected./);
+  });
+  it('Should handle a boolean', () => {
+    const error = createInvalidObservableTypeError(true);
+    expect(error.message).to.match(/^You provided 'true' where a stream was expected./);
+  });
+  it('Should handle a number', () => {
+    const error = createInvalidObservableTypeError(12.34);
+    expect(error.message).to.match(/^You provided '12.34' where a stream was expected./);
+  });
+  it('Should handle null', () => {
+    const error = createInvalidObservableTypeError(null);
+    expect(error.message).to.match(/^You provided 'null' where a stream was expected./);
+  });
+  it('Should handle an Array', () => {
+    const error = createInvalidObservableTypeError([]);
+    expect(error.message).to.match(/^You provided type 'Array' where a stream was expected./);
+  });
+  it('Should handle an object literal', () => {
+    const error = createInvalidObservableTypeError({
+      foo: 'bar',
+      fiz: 'buz',
+    });
+    expect(error.message).to.match(/^You provided type 'Object' where a stream was expected./);
+  });
+
+  /** objects with constructors **/
+  it('Should handle a Map', () => {
+    const error = createInvalidObservableTypeError(new Map());
+    expect(error.message).to.match(/^You provided type 'Map' where a stream was expected./);
+  });
+  it('Should handle a Set', () => {
+    const error = createInvalidObservableTypeError(new Set());
+    expect(error.message).to.match(/^You provided type 'Set' where a stream was expected./);
+  });
+  it('Should handle an Error', () => {
+    const error = createInvalidObservableTypeError(new Error());
+    expect(error.message).to.match(/^You provided type 'Error' where a stream was expected./);
+  });
+  it('Should handle a TypeError', () => {
+    const error = createInvalidObservableTypeError(new TypeError());
+    expect(error.message).to.match(/^You provided type 'TypeError' where a stream was expected./);
+  });
+  it('Should handle a Proxy', () => {
+    const error = createInvalidObservableTypeError(new Proxy(new Map(), {}));
+    expect(error.message).to.match(/^You provided type 'Map' where a stream was expected./);
+  });
+  it('Should handle an Observable', () => {
+    const error = createInvalidObservableTypeError(from<string>('FooBar'));
+    expect(error.message).to.match(/^You provided type 'Observable' where a stream was expected./);
```

Comment:
I understand what this is trying to do, but this error would make me want to scream if I saw it and didn't know what it meant.

```ts
"You provided type 'Observable' where a stream was expected. You can provide an Observable, Promise, Array, AsyncIterable, or Iterable."
```

We probably need to do something where we actually link to some documentation about what could be going on. :thinking:

Or at the very least maybe it needs to say something like "unsupported Observable" or.. something that makes RxJS look less like it's losing it's mind.

Reply:
IMO, it's worth checking the ctor name against `'Observable'` and saying something like:

```
"You provided type 'Observable' (from another package) where a stream was expected. ..."
```

Reply:
thanks for the suggestions. i'll get it done.